Slim/Player/Protocols/HTTP.pm

```diff
@@ -42,16 +49,82 @@ sub new {
 		# return undef;
 	}
 
-	my $self = $class->SUPER::new($args);
+	my $self = $class->SUPER::new($args) or do {
+		$log->error("Couldn't create socket binding to $main::localStreamAddr - $!");
+		return undef;
+	};
 
-	if (defined($self)) {
-		${*$self}{'client'}  = $args->{'client'};
-		${*$self}{'url'}     = $args->{'url'};
-	}
+	${*$self}{'client'}  = $args->{'client'};
+	${*$self}{'url'}     = $args->{'url'};
+	${*$self}{'_class'}  = $class;	
 
 	return $self;
 }
 
+sub close {
+	my $self = shift;
+
+	# call parent's ONLY when new() was made by this class, otherwise
+	# let subclass take care of socket's close (multiple inheritance)
+	$self->SUPER::close(@_) if ${*$self}{'_class'};
+	return unless my $v = delete ${*$self}{'_enhanced'};
```

Comment:
Please rename `$v` to something more speaking.

Reply:
any suggestion? It's just a hash to the structure used to store 'enhanced' internal data/states. I wanted it to be short because it does not have any meaning, except being less verbose

Reply:
see below, $v is really a mute variable for me, it's just a convenience for more compact coding instead of ${*$self}{_enhanced} which gives me headaches.

Reply:
We could call it `$enhanced` for no better idea...

Reply:
I finally went up for $enhanced, it's better than $context b/c $context does not mean much more than $v and is longer, so per our discussion, if I go for a longer name, give it a useful meaning :smile: